File: onebot-sdk/src/main/java/cn/evole/onebot/sdk/response/group/GroupHonorInfoResp.java
package cn.evole.onebot.sdk.response.group;

import com.google.gson.annotations.SerializedName;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Created on 2022/7/8.
 *
 * @author cnlimiter
 */
@Data
public class GroupHonorInfoResp {

    @SerializedName("group_id")
    private long groupId;

    @SerializedName("current_talkative")
    private CurrentTalkative currentTalkative;

    @SerializedName("talkative_list")
    private List<OtherHonor> talkativeList;

    @SerializedName("performer_list")
    private List<OtherHonor> performerList;

    @SerializedName("legend_list")
    private List<OtherHonor> legendList;

    @SerializedName("strong_newbie_list")
    private List<OtherHonor> strongNewbieList;

    @SerializedName("emotion_list")
    private List<OtherHonor> emotionList;

    /**
     * 活跃天数
     */
    @Data
    @NoArgsConstructor
    public static class CurrentTalkative {

        @SerializedName("user_id")
        private long userId;

        @SerializedName("nickname")
        private String nickname;

        @SerializedName("avatar")
        private String avatar;

        @SerializedName("day_count")
        private int dayCount;


    }

    /**
     * 其它荣耀
     */
    @Data
    @NoArgsConstructor
    public static class OtherHonor {

        @SerializedName("user_id")
        private long userId;

        @SerializedName("nickname")
        private String nickname;

        @SerializedName("avatar")
        private String avatar;

        @SerializedName("description")
        private String description;


    }


}

File: onebot/src/main/kotlin/sdk/response/group/GroupHonorInfoResp.kt
package cn.evolvefield.onebot.sdk.response.group

import com.google.gson.annotations.SerializedName
import lombok.Data
import lombok.NoArgsConstructor

@Data
class GroupHonorInfoResp {
    @SerializedName("group_id")
    var groupId = 0L
    @SerializedName("current_talkative")
    private val currentTalkative: CurrentTalkative? = null
    @SerializedName("talkative_list")
    var talkativeList: List<OtherHonor> = mutableListOf()
    @SerializedName("performer_list")
    var performerList: List<OtherHonor> = mutableListOf()
    @SerializedName("legend_list")
    var legendList: List<OtherHonor> = mutableListOf()
    @SerializedName("strong_newbie_list")
    var strongNewbieList: List<OtherHonor> = mutableListOf()
    @SerializedName("emotion_list")
    var emotionList: List<OtherHonor> = mutableListOf()

    /**
     * 活跃天数
     */
    @Data
    @NoArgsConstructor
    class CurrentTalkative {
        @SerializedName("user_id")
        var userId = 0L

        @SerializedName("nickname")
        var nickname = ""

        @SerializedName("avatar")
        var avatar = ""

        @SerializedName("day_count")
        var dayCount = 0
    }

    /**
     * 其它荣耀
     */
    @Data
    @NoArgsConstructor
    class OtherHonor {
        @SerializedName("user_id")
        var userId = 0L

        @SerializedName("nickname")
        var nickname = ""

        @SerializedName("avatar")
        var avatar = ""

        @SerializedName("description")
        var description = ""
    }
}
